internal/topology/variables/clusterclass_variable_validation.go

```diff
@@ -81,15 +87,14 @@ func validateClusterClassVariableName(variableName string, fldPath *field.Path)
 	allErrs := field.ErrorList{}
 
 	if variableName == "" {
-		allErrs = append(allErrs, field.Invalid(fldPath, variableName, "name cannot be empty"))
+		allErrs = append(allErrs, field.Required(fldPath, "variable name must be defined"))
 	}
 
-	// TODO(sbueringer): This should share a constant with controllers/topology/internal/extensions/patches/variables.BuiltinsName.
-	if variableName == "builtin" {
-		allErrs = append(allErrs, field.Invalid(fldPath, variableName, "name cannot be \"builtin\""))
+	if variableName == builtinsName {
+		allErrs = append(allErrs, field.Invalid(fldPath, variableName, fmt.Sprintf("%q is a reserved variable name", builtinsName)))
 	}
 	if strings.Contains(variableName, ".") {
-		allErrs = append(allErrs, field.Invalid(fldPath, variableName, "name cannot contain \".\""))
+		allErrs = append(allErrs, field.Invalid(fldPath, variableName, "variable name cannot contain \".\"")) // TODO: consider if to restrict variable names to RFC 1123
```

Comment:
Should we mention that it's about the DNS subdomain part of RFC 1123 (IsDNS1123Subdomain ?) 
(not sure if there are other relevant parts of the RFC)

Reply:
Only if we actually implement additional restrictions